samples/subsys/usb/dfu/Kconfig

```diff
@@ -4,4 +4,9 @@
 config USB_DEVICE_PID
 	default USB_PID_DFU_SAMPLE
 
+DT_COMPAT_QSPI_NOR := nordic,qspi-nor
```

Comment:
That does not seem right to me, let me block it for now.

Reply:
> grabbed that from `samples/subsys/usb/mass/Kconfig`

I am not sure it would not be too general.  There is also #47782 to fix it on bl5340_dvk.

Reply:
How about having this in https://github.com/zephyrproject-rtos/zephyr/blob/main/drivers/flash/Kconfig.nordic_qspi_nor#L4 directly? We may be able to drop few overrides that way (rather than adding more).

Reply:
That would always enable QSPI driver but that is (I suspect) not desired.

Reply:
> That would always enable QSPI driver but that is (I suspect) not desired.

Why not?  If CONFIG_FLASH=y, than I think its reasonable to enable a FLASH driver if one exists & is enabled in dts.

@carlescufi any thoughts on enabling nordic QSPI by default if enabled in dts?

Reply:
> > That would always enable QSPI driver but that is (I suspect) not desired.
> 
> Why not? If CONFIG_FLASH=y, than I think its reasonable to enable a FLASH driver if one exists & is enabled in dts.

I agree, that's reasonable. But it's better to do it, as @fabiobaltieri already suggested above, by adding `default y` in `Kconfig.nordic_qspi_nor`.

Reply:
@galak mind doing it the way suggested above?

Reply:
> @galak mind doing it the way suggested above?

Done.